Gateway/Request/Invoice/AbstractInvoiceDataBuilder.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Buckaroo\Magento2\Gateway\Request\Invoice;
+
+use Buckaroo\Magento2\Gateway\Request\AbstractDataBuilder;
+use Buckaroo\Magento2\Helper\Data as BuckarooHelper;
+
+abstract class AbstractInvoiceDataBuilder extends AbstractDataBuilder
+{
+    private BuckarooHelper $buckarooHelper;
+    protected int $numberOfInvoices;
+    protected bool $capturePartial = true;
+    protected float $currentInvoiceTotal;
+
+    public function __construct(BuckarooHelper $buckarooHelper)
+    {
+        $this->buckarooHelper = $buckarooHelper;
+    }
+
+    public function initialize(array $buildSubject): array
+    {
+        $data = parent::initialize($buildSubject);
+
+        $order = $this->getOrder();
+
+        $totalOrder = $order->getBaseGrandTotal();
+        $this->numberOfInvoices = $order->getInvoiceCollection()->count();
+        $this->currentInvoiceTotal = 0;
+
+        // loop through invoices to get the last one (=current invoice)
+        if ($this->numberOfInvoices) {
+            $oInvoiceCollection = $order->getInvoiceCollection();
+
+            $i = 0;
```

Comment:
what's the use of this `$i`?

Reply:
$i is used to skip all invoice until the last one 

if (++$i !== $numberOfInvoices) {
                    continue;
}

The code was copied from app/code/Buckaroo/Magento2/Model/Method/AbstractMethod.php:1606

On Buckaroo/Magento2/Gateway/Request/InvoicedArticlesDataBuilder.php:37 I used filters to get the last one: 

$currentInvoice = $invoiceCollection
            ->addAttributeToSort('created_at', 'DSC')
            ->setPage(1, 1)
            ->getFirstItem();

Should I use the same for AbstractInvoiceDataBuilder?